From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct mlx5_cmd_work_ent
Patch-mainline: never, kabi
References: kabi

In networking-stable-20_05_27, commit 17d00e839d3b (net/mlx5: Add
command entry handling completion) added a member to struct
mlx5_cmd_work_ent. This indeed changed the layour of the structure and
the kABI checker complains now.

struct mlx5_cmd_work_ent is internal to mlx5 driver, so workaround the
breakage by the usual __GENKSYMS__ trick.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/mlx5/driver.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/mlx5/driver.h
+++ b/include/linux/mlx5/driver.h
@@ -761,7 +761,9 @@ struct mlx5_cmd_work_ent {
 	struct delayed_work	cb_timeout_work;
 	void		       *context;
 	int			idx;
+#ifndef __GENKSYMS__
 	struct completion	handling;
+#endif
 	struct completion	done;
 	struct mlx5_cmd        *cmd;
 	struct work_struct	work;
